lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
     class DirectObjectReference
       def initialize(object)
         @object = object
       end
 
+      # The objects desription (`#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Not relevant to a direct object reference.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Class
       def target
         @object.class
       end
```

Comment:
Also, it's possible to go from an instance to a class (just call `.class`!) but it's not possible to go from a class to the original instance (unless you do gnarly object space stuff...) so converting this to its class loses information in the sense that the caller has no easy way to get the object from it.  OTOH, if we return the object from `target` and the user wants a class, then they can do:

``` ruby
target = reference.target
target = target.class unless Module === target
```